magstock/templates/regextra.html

```diff
@@ -0,0 +1,116 @@
+
+<script type="text/javascript">
+    if (window.BADGE_TYPES) {
+        BADGE_TYPES[0].description = 'Includes three day admission and campground fees';
+        BADGE_TYPES[1].description = 'T-shirt level plus additional swag. (Available until {{ c.SUPPORTER_DEADLINE|datetime:'%b %d' }}. Limited quantities.)';
+        {% if c.SHIRT_LEVEL in c.PREREG_DONATION_TIERS %}
+            BADGE_TYPES.splice(1, 0, {
+                title: 'Plus T-Shirt: ${{ c.BADGE_PRICE|add:c.SHIRT_LEVEL }}',
+                description: 'Attendee level plus a MAGStock themed t-shirt',
+                extra: {{ c.SHIRT_LEVEL }}
+            });
+        {% endif %}
+        $(function () {
+            $.field('amount_extra').parents('.form-group').hide();
+        });
+    }
+    var showOrHideAllergies = function () {
+        setVisible('#allergies', $.field('purchased_food').is(':checked'));
+    };
+    var showOrHideComingWith = function () {
+        var showComingWith = ($.val('coming_as') === {{ c.TENT_LEADER }} || $.val('coming_as') === {{ c.TENT_FOLLOWER }});
+        setVisible('#comingwith', showComingWith);
+        if (showComingWith) {
+            var isTentLeader = $.val('coming_as') === {{ c.TENT_LEADER }};
+            setVisible($('#comingwith :text').prop('disabled', isTentLeader), !isTentLeader);
+            setVisible($('#comingwith textarea').prop('disabled', !isTentLeader), isTentLeader);
+            $('#comingwith label').text(isTentLeader ? 'Who is in your group?' : 'Who is your Tent Leader?');
+        }
+    };
+    $(function () {
```

Comment:
These next two lines are how we insert fields in specific places in the form.  Again, this is pretty standard jQuery, but we could theoretically implement custom functions for this as well, like ``moveToBefore('#food', 'staffing')`` and ``moveToAfter('#extra-fields', 'zip_code')`` if people find that more intuitive.  Personally I prefer the raw jQuery, but that might just be me.